### PR TITLE
Add PHP 7.4

### PR DESCRIPTION
**Pull request by EtienneM**
Three extensions are still missing. But maybe we can already release PHP 7.4 without them and I keep working on the 3 last extensions?

I tested PHP 7.4 here: https://sample-php-base.scalingo.io/

- [x] apcu
- [x] blackfire: our script is inspired by Heroku's script (https://github.com/heroku/heroku-buildpack-php/blob/master/support/build/extensions/no-debug-non-zts-20121212/blackfire). It seems Blackfire changed the way they distribute the PHP extension. We should wait for Heroku's modification. https://blackfire.io/docs/up-and-running/installation
- [x] ds
- [x] igbinary
- [x] imagick
- [x] libevent: is not maintained since a long time. Shouldn't we remove it? https://pecl.php.net/package/libevent
- [x] lua: Error `configure: error: Please reinstall the lua distribution - lua.h should be in <lua-dir>/include/`. But this extension is not present in the last major version for scalingo-18. Is it removable?
- [x] mcrypt: removed since PHP 7.2. Shouldn't we remove it? https://www.php.net/manual/fr/intro.mcrypt.php
- [x] memcache: is not maintained since a long time. Shouldn't we remove it? https://pecl.php.net/package/memcache
- [x] memcached
- [x] mongodb
- [x] mongo-legacy: Shouldn't we remove it as its name suggest?..
- [ ] phalcon: instructions changed, need to update the script: https://packagecloud.io/phalcon/stable
- [x] redis
- [x] sundown: is not maintained since a long time. Shouldn't we remove it? https://pecl.php.net/package/sundown

### CODE REVIEW

**Comment by EtienneM** [comment]
I think we never packaged the extensions in the environment of the stack scalingo-18 because the issues I have now are because of this.

**Comment by Soulou** [comment]
What's the state of phalcon and blackfire? This last one is quite annoying, as it's used by customers quite a lot

File: bin/compile [hunk 1]
```diff
@@ -276,8 +280,10 @@ fi
 mkdir -p "conf"
 cp "$basedir/util/autotune.php" "conf/autotune.php"
 cp "$basedir/../conf/nginx/base.conf.erb" "conf/nginx.conf.erb"
-# If PHP 7.3, custom config file to improve logging
-if [ "$(php_api_version)" = "20180731" ] ; then
+# Fix dirty logging on stdout/stderr (WARNING child with PID said on stderr). It
+# has been fixed in PHP 7.3 (https://github.com/php/php-src/pull/2458) with a
+# custom php-fpm parameter: `decorate_workers_output`
+if [[ "$(php_api_version)" > "${PHP_MODULE_API_VERSIONS["7.2"]}" ]] ; then
```
Comment (Soulou) [inline bin/compile]: Isn't it `-gt` ?

File: conf/buildpack.conf [hunk 2]
```diff
@@ -16,10 +16,19 @@ PHP_MODULE_API_VERSIONS["7.0"]="20151012"
...
+# Libmemcached version to build the memcached extension
+libmemcached_version="1.0.18"
+# Deprecated since PHP 7.2 https://www.php.net/manual/en/intro.mcrypt.php
 mcrypt_version="2.5.8"
+# As of PHP 7.3.0 it is automatically included: https://www.php.net/manual/en/zip.installation.php
 zip_version="1.5.2"
 gmp_version="6.1.2"
 tidy_version="5.6.0"
 coreutils_version="8.30"
+# Mandatory for multibytes starting with PHP 7.4
```
Comment (Soulou) [inline conf/buildpack.conf]: multibytes strings no?

**Comment by EtienneM** [comment]
> What's the state of phalcon and blackfire? This last one is quite annoying, as it's used by customers quite a lot

As stated in the first comment:

> blackfire: our script is inspired by Heroku's script (https://github.com/heroku/heroku-buildpack-php/blob/master/support/build/extensions/no-debug-non-zts-20121212/blackfire). It seems Blackfire changed the way they distribute the PHP extension. We should wait for Heroku's modification. https://blackfire.io/docs/up-and-running/installation

Our current script does not work at all because we try to download an archive which does not exist anymore due to their change in the way they distribute the extension.

**Comment by Soulou** [comment]
Can't we find how to do it without the help of Heroku?

**Comment by EtienneM** [comment]
Yes we can!

File: bin/compile (hunk 1 above)
Comment (EtienneM) [reply bin/compile]: `-gt` is for number

`>` works for string. it's a bash-ism

Comment (Soulou) [reply bin/compile]: But in shell, it's the same actually, [ "3" -gt "2" ] -> returns 0 :)

Comment (EtienneM) [reply bin/compile]: Well both works then ^^ I'll switch to `-gt` as you seem more keen on this one

**Comment by EtienneM** [comment]
Blackfire was far easier than I initially expected :)

**Comment by Soulou** [comment]
For LUA it has been added recently https://doc.scalingo.com/languages/php/start#native-php-extensions, was it related to a customer need?

**Comment by EtienneM** [comment]
Customer who asked for PHP Lua extension: https://app.intercom.io/a/apps/w4oogu7s/inbox/inbox/search/conversations/22764901611?assigneeIdentifier=all&q=PHP%20lua

**Comment by EtienneM** [comment]
Actually Phalcon is not working with PHP 7.4 (https://github.com/phalcon/cphalcon/issues/14569#issuecomment-560162133) and may never work for this version of PHP if I read the arguments for PHP 7.3 (https://github.com/phalcon/cphalcon/issues/13847).

Hence I think our support of PHP 7.4 is complete :)

**Comment by Soulou** [comment]
LGTM

**Pull request merged**